### PR TITLE
add recipe for compas_libigl

### PR DESCRIPTION
**Pull request by tomvanmele**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/compas_libigl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  host:
+    - pip
+    - python
+    - pybind11
+  run:
+    - python
+    - eigen <3.4
+    - boost-cpp
+    - nptyping
+    - typing_extensions
```
Comment (gonzalocasas) [inline recipes/compas_libigl/meta.yaml:47]: Should this also require `igl` ?

Comment (tomvanmele) [reply recipes/compas_libigl/meta.yaml:47]: we don't use "their" binding if that is what you are asking

**Comment by tomvanmele** [comment]
@conda-forge-admin ready for review

**Comment by tomvanmele** [comment]
@conda-forge-admin please ping team

File: recipes/compas_libigl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+test:
+  imports:
+    - compas_libigl
+
+about:
+  home: https://github.com/compas-dev/compas_libigl
+  license: MPL-2.0
+  license_file: LICENSE.MPL2
```
Comment (xhochy) [inline recipes/compas_libigl/meta.yaml:62]: Please also add the license files for libigl and pybind11

File: recipes/compas_libigl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  run:
+    - python
+    - numpy
+    - eigen <3.4
+    - boost-cpp
+    - compas
+    - nptyping
+    - typing_extensions
+
+test:
```
Comment (xhochy) [inline recipes/compas_libigl/meta.yaml:55]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/compas_libigl/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{% set name = "compas_libigl" %}
+{% set version = "0.2.2" %}
+{% set pybind11_version = "2.10.4" %}
+{% set libigl_version = "2.2.0" %}
+{% set hash = "e3ee9a9915cb05559ec4c71885ae91b635b28b4ee75753a43bc1eddadc9624da" %}
+{% set pybind11_hash = "832e2f309c57da9c1e6d4542dedd34b24e4192ecb4d62f6f4866a737454c9970" %}
+{% set libigl_hash = "b336e548d718536956e2d6958a0624bc76d50be99039454072ecdc5cf1b2ede5" %}
```
Comment (xhochy) [inline recipes/compas_libigl/meta.yaml:7]: Please don't have Jinja indirections for these hashes.
```suggestion
```

File: recipes/compas_libigl/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/compas-dev/compas_libigl/archive/v{{ version }}.tar.gz
+    fn: compas_libigl-{{ version }}.tar.gz
+    sha256: {{ hash }}
+  - url: https://github.com/pybind/pybind11/archive/v{{ pybind11_version }}.tar.gz
```
Comment (xhochy) [inline recipes/compas_libigl/meta.yaml:17]: It would be preferable to use the pybind11 from codna-forge. Can you file an issue upstream please?

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by tomvanmele** [comment]
@xhochy

* jinja hash variables have been removed
* pybind11 github dependency has been removed
* test section updated

the license included is the one of libigl.
should i include two separate licenses (one for the repo and one for libigl)?
how does one do that?

**Comment by xhochy** [comment]
> should i include two separate licenses (one for the repo and one for libigl)?

Yes, you can make the licenses a list:

```
license_file:
  - LICENSE.MPL2
  - LICENSE.igl
```

**Comment by tomvanmele** [comment]
@xhochy all done, thanks!

**Review by xhochy — APPROVED** [review]

**Pull request merged**